EESSI Governance (#456)
EESSI/docs: docs/governance/governance.md

<!-- 
A project charter discusses _what it is and why it exists_, a governance discusses _how it operates_.
-->

# Project Governance

## 1. Guiding Principles
<!-- Optional section to state high-level principles like openness, meritocracy, consensus, etc. -->

> The value of EESSI grows exponentially with two things: the amount of systems that make the EESSI software stack available, and the amount of software that is available in the EESSI software stack. Thus, the first goal of this governance is to make sure everyone in the community feels sufficiently included so that they are willing to contribute to EESSI (rather than build their own solution).

Reviewer: What about CPU targets we support? I see that as a 3rd dimension?

> The second goal of this governance is to make clear how and by whom decisions in EESSI are taken. This is because trust in this process is important, both to infrastructure providers making the EESSI software stack is available on their systems, as well as by end-users of the software. Note that this concerns both large decisions, such as which architectures are supported in EESSI, as well as small decisions, such as whether to accept a certain contribution to add software to EESSI.

Reviewer: "EESSI software stack" -> EESSI

"such as whether to accept a certain contribution to add software to EESSI" -> "such as making a specific software package in EESSI"

Author: Changed

> To achieve both goals, our governance is based on the meritocracy governance model.

Reviewer: Link to something that explains what "meritocracy" is? Maybe https://en.wikipedia.org/wiki/Meritocracy ?

## 2. Roles and Responsibilities

Below, the roles and responsibilities related to the EESSI project are discussed. The group of people with a common role will be referred to as a Team in the remainder of this document. Exceptions are Contributors, System administrators of systems providing EESSI and End-users (these are not considered to be Teams). Each individual in a Team will be referred to as a Team Member.

> ### 2.1 EESSI Github organization & repository owners

Reviewer: ```suggestion
### 2.1 Owners of the [EESSI GitHub organization](https://github.com/EESSI) & repositories

Author: Fixed

<!-- Define who contributors are and what they can do (e.g., file issues, submit PRs). -->

> EESSI Github organization & repository owners are those individuals with `owner` rights on the EESSI Github organization or one of it's associated repositories.

Author: Fixed

EESSI Github organization & repository owners shall be responsible for setting permissions on the code repositories, compliant to the defined roles and responsibilities. They are also responsible for managing Github Apps for the EESSI build bots.

### 2.2 EESSI Github repository maintainers
EESSI Github repository maintainers are individuals with write access (one or more) of the EESSI Github repositories.

> EESSI Github repository maintainers shall be responsible for reviewing and merging PRs.

Reviewer: ```suggestion
EESSI GitHub repository maintainers are responsible for reviewing and merging pull requests (contributions).

Author: Fixed

### 2.3 EESSI Infrastructure maintainers
EESSI Infrastructure maintainers are those individuals that maintain the EESSI CVMFS Stratum 0, one of the _public_ EESSI CVMFS Stratum 1's, build infrastructure (hosting the EESSI build bot) and the SMEE server.

> Infrastructure maintainers shall be responsible for monitoring and maintaining their respective infrastructure, and provide access to those who needed it according to the Roles and Responsibilities described here. Note that maintainers of build infrastructure have the right to limit access to a _subset_ of the Deployers and Builders as described in 2.4 and 2.5 if they so desire. Furthermore, they are _not_ allowed to give access to others that are not part of the EESSI Infrastructure maintainers, Deployers or Builders teams.

Author: Changed

### 2.4 Builders
Builders are those individuals that have permissions to build software through one or more of the EESSI build bots.

Builders are responsible for triggering builds for contributors who want to add software to the EESSI software stack. They are also responsible for checking the PR, so as to avoid execution of malicious code on the EESSI build infrastructure.

> ### 2.5 Deployers

Author: It's not a super common term - I find no clear hits on Google on CernVM-FS Librarian. Some on CernVM-FS software librarian, but the only complete match comes from UGent ;-)

We also coined "Ingestors" in the meeting. I personally prefer that, as it's more descriptive. As discussed, I'll add it as a separate role, as not all deployers are ingestors. Essentially, deployers have the authority to deploy tarballs to S3, as described on https://www.eessi.io/docs/adding_software/overview/, and ingestors have the rights to ingest staged tarballs into the software.eessi.io repository (again, as described on https://www.eessi.io/docs/adding_software/overview/ )

## 4. Meetings and Communication
<!-- Define how and where the team communicates, and how meetings (if any) are run. -->
While each Team is allowed to use the communication channels that their Team jointly agrees upon, the following channels are suggested
- GitHub / GitLab issues
- Slack
- (Periodic or incidental) video calls

## 5. Onboarding and Offboarding
This Section applies to the Teams described in Section 2.

### 5.1 Adding Team Members
<!-- Describe the process for adding new members. -->
Teams decide themselves how large their teams should ideally be. They also decide the procedure to add new Team members. The procedure should reflect the sensitivity of the position, i.e. people with certain roles have the ability to (intentionally or unintentionally) compromise the integrity of the repository. For such roles, the procedure should make sure candidates are extensively vetted and trusted by both the Team as well as the EESSI community as a whole.

As with all decisions the decision to add a Team member is subject to article 3.4.

### 5.2 Removing Team Members
<!-- Describe under what conditions someone may be removed (e.g., inactivity, conduct). -->

> Teams decide themselves decide the procedure to remove new Team members. As for the procedure of adding Team Members, the procedure to remove Team Members should reflect the sensitivity of the position.

Reviewer 2: ```suggestion
Teams themselves decide the procedure to remove Team members. As for the procedure of adding Team Members, the procedure to remove Team Members should reflect the sensitivity of the position.
```

Anything may be brought to a vote during the meeting. However, if known up-front, issues that need to be voted on are announced at least one week prior to the meeting, to give Members the opportunity to cast an electronic vote if they cannot attend.

To amend the Charter or Governance, a two-thirds majority of all Steering Committee Members needs to be in favour of the amendment. This is irrespective of the number of Members who have actually cast a vote, or are present at a meeting. There is no quorum for votes on ammendments to the Charter or Governance.

### 6.4 Meetings

The Steering Committee will meet as needed, but at least once per quarter. The Chair is responsible for organizing the meeting. Any Member can request the Chair to organize a meeting. Meetings are announced at least 2 weeks in advance, unless _all_ Members have agreed that a meeting on shorter notice is required.

## 7. Code of Conduct
<!-- Reference your Code of Conduct or provide a brief summary. -->
TODO: This project follows the [Contributor Covenant](https://www.contributor-covenant.org/) Code of Conduct.

## 8. Contribution Agreement

> TODO: Should refer to some Contribution Agreement. Is contributing only possible after signing this agremeent? If so, that should be stated here

Author: Hm, that sounds attractive, since it means we don't need a lawyer to draw up something for us :)

### 2.8 System administrators of systems providing EESSI

System administrators of systems providing EESSI are administrators of systems (such as, but not limited to cloud and HPC systems) that make the EESSI software stack available to their users.

System administrators of systems providing EESSI should make sure that their system does not put any disproportional load on the public EESSI CernVM-FS infrastructure. Typically, this means that they should adhere to best practices for CernVM-FS caching in a way that fits the size and nature of their system (e.g. by setting up a private CernVM-FS Stratum 1, a proxy, and/or a properly sized local cache).

System administrators have to agree to the EESSI [Terms of Use](terms_of_use.md).

### 2.9 End-Users

End-Users are individuals that use any of the software provided by the EESSI software stack.

End users have to agree to the EESSI [Terms of Use](terms_of_user.md).

## 3. Decision-Making

> This section applies to the decision making procedure for the Teams discussed in Section 2. The Steering Committee has its own Decision-Making procedure. 

Author: Changed to "The Steering Committee has its own Decision-Making procedure (defined in Section 6.3)."

### 3.1 Consensus Seeking
<!-- Describe how decisions are reached: rough consensus, lazy consensus, etc. -->

> Decisions are made by seeking consensus. Each Team member is responsible for deciding whether an issue or decision is significant or sensitive enough to warrant discussions within their Team. If so, they should bring it up through the agreed-upon channels (see Section 4).

Author: Changed

### 3.2 Voting
<!-- Detail when voting is used, who can vote, quorum rules, and thresholds. -->

> There is no formal voting for Teams, for two reasons:

Author: Already fixed by your change

1. The Teams may be too big to organize voting in a quick, practical way.
2. The obligation to take part in votes may discourage people to become part of these Teams, as it may be seen as a burden.

Voting may nonetheless be used as a way to achieve consensus. For example, asking (a relevant subset of) the Team to vote may be one way to determine the majority opinion. Then, it can be discussed if the majority vote is an acceptable outcome to the Team. If so, consensus is achieved.

### 3.3 Deadlock Resolution
<!-- Define how disputes are escalated and resolved. -->

> If consensus cannot be reached within the Team, the Steering Committee can take the decision for them. This can be requested by the Team, or done at the initiative of the Steering Committee (if they feel the lack of a decision is preventing the project from moving forward).

Author: I don't think it needs to be that detailed: if you're on a team, you _will_ find a way to contact the SC if you feel this is really needed..

Also: this should ideally never happen, and if it happens, it'll be very rare. I imagine the SC will enter into a discussion with the team anyway to get clarification (what is the deadlock about? Does everyone even agree that there is a deadlock? What are the arguments for both sides?). If anyone, through any means contacts the SC for deadlock resolution, I think that's good enough to start such a discussion. I don't think this is a mechanism that can (or will) be abused, and unless you fear that it will, I see little reason to make this more specific.

> If consensus cannot be reached within the Team, the Steering Committee can take the decision for them. This can be requested by the Team, or done at the initiative of the Steering Committee (if they feel the lack of a decision is preventing the project from moving forward).

Reviewer: OK, makes sense to leave this open, we can adjust later should it be needed.

### 3.4 Overturning decisions
<!-- Steering committee overruling a group? -->
The goal is for Teams to operate autonomously as much as possible. 

However, in exceptional cases, the Steering Committee _can_ overrule a decision made by any of the Teams. This power should only be used as a last resort. Examples of when this may be considered is if the Steering Committee feels that a decision endangers the integrity or sustainability of the project, or goes against the policies the Steering Committee has defined. Even then, the Steering Committee should first engage in a discussion with a Team to see if consensus can be achieved on changing the decision.

## 4. Meetings and Communication
<!-- Define how and where the team communicates, and how meetings (if any) are run. -->
While each Team is allowed to use the communication channels that their Team jointly agrees upon, the following channels are suggested
- GitHub / GitLab issues
- Slack
- (Periodic or incidental) video calls

## 5. Onboarding and Offboarding
This Section applies to the Teams described in Section 2.

### 5.1 Adding Team Members
<!-- Describe the process for adding new members. -->

> Teams themselves decide how large their teams should ideally be. They also decide the procedure to add new Team members. The procedure should reflect the sensitivity of the position, i.e. people with certain roles have the ability to (intentionally or unintentionally) compromise the integrity of the repository. For such roles, the procedure should make sure candidates are extensively vetted and trusted by both the Team as well as the EESSI community as a whole.

Author: Hm, I don't think it's a strong word at all - and also according to Wikipedia at least it's a term that's also used for checking a potential future employee for example. That's the type of vetting we would mean here https://en.wikipedia.org/wiki/Vetting . Maybe @ocaisa as native speaker can comment on whether vetted typically implies a _heavy_ background check - or just any background check?

To my understanding, even just getting some references on a person could be considered vetting.

> Teams themselves decide how large their teams should ideally be. They also decide the procedure to add new Team members. The procedure should reflect the sensitivity of the position, i.e. people with certain roles have the ability to (intentionally or unintentionally) compromise the integrity of the repository. For such roles, the procedure should make sure candidates are extensively vetted and trusted by both the Team as well as the EESSI community as a whole.

Reviewer 2: It is used in official contexts, something less intimidating would be `assessed`

- The Steering Committee ideally consists of an odd number of Members
- No more than 30% of the Members should be employed by commercial entities
- The number of Members working for the same company should be limited to 1
- The number of Members from the same country should be limited to 2
- The composition of the Steering Committee should reflect the EESSI community

Note that if Members change jobs, or move to another country, some of the above criteria that were taken into account during selection may no longer be satisfied. It is up to the Steering Committee to decide if this is problematic, and if so, try to restore balance by requesting someone to resign.

All new Members will appoint an alternate that may represent them and vote on their behalf in case they are unavailable. If a Member resigns or is voted out from the Steering Committee, their alternate immediately loses any rights they had as an alternate.

> A Chair will be appointed by the Steering Committee to serve for a one-year term.

Author: Good point. I wouldn't do automatic extension with a year, I think it's good if this is a conscious decision. I also realize that we haven't defined what the role of the chair is.

```suggestion
A Chair will be appointed by the Steering Committee to serve for a one-year term. The Chair is responsible for organizing and leading the Steering Committee meetings (e.g. preparing and sending around an agenda, bringing items to a vote, counting the votes, ensuring meeting notes are kept and made available to the Steering Committee Members, etc).

The Chair can appoint another Steering Committee member to serve as acting Chair for a given meeting or period.

If there is no Chair (e.g. because the Chair's term was completed, or the Chair stepped down from the Steering Committee), any Committee Member can call for and organize a new meeting or electronic vote to appoint a new Chair. Whenever there is no Chair and there is a meeting, the first order of business will be to elect a new Chair.
```

In case of a tie, the issue is discussed again, and a second vote is taken. All Members that voted in the first round need to have the opportunity to vote in the second round, which may mean that the vote needs to be postponed to a next meeting (e.g. because one or more Members voted by e-mail, and are not present at the meeting).

In case the quorum of 75% is not reached, the vote is postponed.

A fully electronic vote (i.e. all Members vote electronically) can be organized by the Chair, in order to prevent having to delay a vote to the next meeting. Such a fully electronic vote is only valid once _all_ Members have cast their vote, or have explicitly stated that they abstain from voting.

Anything may be brought to a vote during the meeting. However, if known up-front, issues that need to be voted on are announced at least one week prior to the meeting, to give Members the opportunity to cast an electronic vote if they cannot attend.

> To amend the Charter or Governance, a two-thirds majority of all Steering Committee Members needs to be in favour of the amendment. This is irrespective of the number of Members who have actually cast a vote, or are present at a meeting. There is no quorum for votes on amendments to the Charter or Governance.

Author: TODO: make the 'all' italic

> To amend the Charter or Governance, a two-thirds majority of all Steering Committee Members needs to be in favour of the amendment. This is irrespective of the number of Members who have actually cast a vote, or are present at a meeting. There is no quorum for votes on amendments to the Charter or Governance.

Reviewer: Indeed, I overlooked the 2/3rd majority of *all* SC members.

It's clear now why requiring quorum for this doesn't make much sense, so all good

### 6.4 Meetings

The Steering Committee will meet as needed, but at least once per quarter. The Chair is responsible for organizing the meeting. Any Member can request the Chair to organize a meeting. Meetings are announced at least 2 weeks in advance, unless _all_ Members have agreed that a meeting on shorter notice is required.

## 7. Code of Conduct
The EESSI Code of Conduct can be found [here](code_of_conduct.md).

## 8. Contribution Agreement

> All commit messages for contributions should be signed by the contributor, i.e. they should contain the statement: "Signed-off-by: firstname lastname <email>". With that signature, the contributor agrees to the [Developer Certificate of Origin](https://developercertificate.org/).

Author: Ah, that's a solution too.

> All commit messages for contributions should be signed by the contributor, i.e. they should contain the statement: "Signed-off-by: firstname lastname <email>". With that signature, the contributor agrees to the [Developer Certificate of Origin](https://developercertificate.org/).

Reviewer: Our stance with writing up the governance is that we should document how things are run now.
We've never required a DCO before, and hence it should definitely be done through a follow-up PR, after CI & documentation is in place (and probably after a trial period with selected people).

So I'm all for removing this for now, and opening an issue to put this in place later (we can use it as a test case for our voting procedure to make changes to governance too btw)

> All commit messages for contributions should be signed by the contributor, i.e. they should contain the statement: "Signed-off-by: firstname lastname <email>". With that signature, the contributor agrees to the [Developer Certificate of Origin](https://developercertificate.org/).

Author: Ok, I left it as a comment, so we don't forget / can easily put it back.

<!-- 
A project charter discusses _what it is and why it exists_, a governance discusses _how it operates_.
-->

# Project Governance

As stated in the [Charter](charter.md), this project governance _only_ covers collaboration on the `software.eessi.io` CernVM-FS repository, it does not cover other CernVM-FS repositories under the eessi.io namespace. In other words, when the sections below mention the EESSI CernVM-FS repository, this implies the `software.eessi.io` repository specifically. Thus, this governance only addresses contributing to, building for, deploying to, ingesting in, providing, or using the `software.eessi.io` repository.

> The only exception to this is the Steering Committee, which has authority over all the `eessi.io` CernFM-FS repositories.

Author: ```suggestion
The only exception to this is the Steering Committee, which has authority over all the `eessi.io` CernVM-FS repositories.
```

- No more than 30% of the Members should be employed by commercial entities;
- The number of Members working for the same company should be limited to 1;
- The number of Members from the same country should be limited to 2;
- The composition of the Steering Committee should reflect the EESSI community;

Note that if Members change jobs, or move to another country, some of the above criteria that were taken into account during selection may no longer be satisfied. It is up to the Steering Committee to decide if this is problematic, and if so, try to restore balance by requesting someone to resign.

All new Members will appoint an alternate that may represent them and vote on their behalf in case they are unavailable. If a Member resigns or is voted out from the Steering Committee, their alternate immediately loses any rights they had as an alternate.

#### 6.2.3 Chair

A Chair will be appointed by the Steering Committee to serve for a one-year term.

### 6.3 Voting by the Steering Committee {#voting-by-sc}

#### 6.3.1 Majority vote

General decisions are taken by majority vote, provided that at least 75% of the Members cast a vote.

#### 6.3.2 Ways of voting

> Votes can be cast either in person, participating electronically, or via electronic vote (such as by e-mail).

Reviewer: Wording is a bit weird here: in-person meeting, virtual meeting, electronic vote (email, approved review in GitHub pull request, ...)

#### 6.3.4 Abstaining

Members can also abstain from voting, in which case they neither count towards the quorum or calculating the majority. Example: in a committee with 7 members, if 3 members vote in favor, 2 vote against, and one abstains from voting, the quorum of 6 (75% of 7 members, rounded up) is not met.

#### 6.3.5 Ties

In case of a tie, the issue is discussed again, and a second vote is taken. All Members that voted in the first round need to have the opportunity to vote in the second round, which may mean that the vote needs to be postponed to a next meeting (e.g. because one or more Members voted by e-mail, and are not present at the meeting).

#### 6.3.6 Failing to reach quorum

In case the quorum of 75% is not reached, the vote is postponed.

#### 6.3.7 Fully electronic vote

A fully electronic vote (i.e. all Members vote electronically) can be organized by the Chair, in order to prevent having to delay a vote to the next meeting. Such a fully electronic vote is only valid once _all_ Members have cast their vote, or have explicitly stated that they abstain from voting.

#### 6.3.8 Voting topics

> Anything may be brought to a vote during the meeting. However, if known up-front, issues that need to be voted on are announced at least one week prior to the meeting, to give Members the opportunity to cast an electronic vote if they cannot attend.

Author: ```suggestion
Anything may be brought to a vote during the meeting. However, if known up-front, issues that need to be voted on are announced by the Chair at least one week prior to the meeting, to give Members the opportunity to cast an electronic vote if they cannot attend.
```

<!-- 

> A project charter discusses _what it is and why it exists_, a governance discusses _how it operates_.

Author: Honestly, we can scrap this if you don't like it. It was meant to provide some minimal explanation to the other steering committee members (those who didn't do research on project charters and governance) about why these are two separate things (and seperate documents, even).

- A Member may be voted out. In this case, the vote needs to be unanimous between the other Members. This procedure is intended to provide a mechanism for addressing extraordinary circumstances where a members behavior or actions are deemed severely detrimental to the committee's function or reputation, and/or to the EESSI project.
- If a Member is inactive for 6 months and does not reply to communication from the rest of the Steering Committee, the Member may be voted out by the other Steering Committee members. In this case, [regular voting rules](#voting-by-sc) apply.

#### 6.2.2 New members

New Members must be recommended by one or more Steering Committee members.

The Steering Committee will make the recommendation known to the community, and collect feedback on the recommendation.
The Steering Committee then weighs the feedback, and votes on whether to accept the new member.

In the selection process, the Steering Committee will consider the following:

- The Steering Committee ideally consists of an odd number of Members;
- No more than 30% of the Members should be employed by commercial entities;
- The number of Members working for the same company should be limited to 1;

> - The number of Members from the same country should be limited to 2;

Author: Changed. Also added the motivation for each of these - I think that's useful, as it allows a steering committee to judge _when_ it is a problem, and when it isn't (e.g. when members change jobs, see below).

- The composition of the Steering Committee should reflect the EESSI community;

Note that if Members change jobs, or move to another country, some of the above criteria that were taken into account during selection may no longer be satisfied. It is up to the Steering Committee to decide if this is problematic, and if so, try to restore balance by requesting someone to resign.

> All new Members will appoint an alternate that may represent them and vote on their behalf in case they are unavailable. If a Member resigns or is voted out from the Steering Committee, their alternate immediately loses any rights they had as an alternate.

Reviewer: Maybe we should make this optional, and even leave the door open for temporary alternate (for 1 meeting)?

### 2.1 Owners of the `EESSI` GitHub organization & repositories
<!-- Define who contributors are and what they can do (e.g., file issues, submit PRs). -->
The owners of the [EESSI GitHub organization](https://github.com/EESSI) & repositories are those individuals with `owner` rights on the EESSI GitHub organization or one of its associated repositories.

EESSI GitHub organization & repository owners are responsible for setting permissions on the code repositories, compliant to the defined roles and responsibilities. They are also responsible for managing GitHub Apps for the EESSI build bots.

### 2.2 EESSI GitHub repository maintainers
EESSI GitHub repository maintainers are individuals with write access to (one or more of) the EESSI GitHub repositories.

EESSI GitHub repository maintainers are responsible for reviewing and merging PRs.

### 2.3 EESSI Infrastructure maintainers

> EESSI Infrastructure maintainers are those individuals that maintain the Central Server for the EESSI CernVM-FS repository (the CernVM-FS Stratum 0), the _public_ mirror servers (CernVM-FS Stratum 1's), and EESSI build infrastructure (e.g. hosting the EESSI build bot).

Reviewer: ```suggestion
EESSI Infrastructure maintainers are those individuals that maintain the Central Server for the EESSI CernVM-FS repository (the CernVM-FS Stratum 0), the _public_ mirror servers (CernVM-FS Stratum 1's), the EESSI build infrastructure (e.g. hosting the EESSI build bot), and/or manage the domain registration(s) connected to EESSI.
```

Builders are responsible for reviewing contributions. They should, for example, check that contributions adhere to the contribution guidelines, that a contribution will not trigger execution of any unintended or unwanted code, etc. They are also responsible for triggering builds for contributors who want to add software to the EESSI software stack.

### 2.5 Deployers
Deployers are those individuals that have permissions to deploy software through one or more of the EESSI build bots to a central location (e.g. S3 bucket), i.e. they are essentially responsible for step 2 as it is described in the [Builders](#builders) Section.

Deployers should check the file list of the produced tarballs (as it is reported by the build bot) for unexpected items. They should also check if all officially supported hardware targets have been built for.

### 2.6 Ingesters
Ingesters are those individuals that have permissions to ingest a tarball into the EESSI CernVM-FS repository, i.e. they are essentially responsible for step 3 as it is described in the [Builders](#builders) Section.

Ingesters have merge permissions on the (private) `EESSI/staging` GitHub repository.

> Ingesters should check the contents of the tarball as it is reported in the pull requests in the `EESSI/staging` repository. This essentially provides a second check (next to that done by Deployers) on the contents of the tarball.

Reviewer 3: Recently, we started using `staging_bundles`. Maybe `EESSI/staging*` is more flexible?

Reviewer: ```suggestion
Ingesters should check the contents of the tarball as it is reported in the pull requests in the staging repository. This essentially provides a second check (next to that done by Deployers) on the contents of the tarball.
```

### 2.8 System administrators of systems providing EESSI

System administrators of systems providing EESSI are administrators of systems (such as, but not limited to cloud and HPC systems) that make the EESSI software stack available to their users.

System administrators of systems providing EESSI should make sure that their system does not put any disproportionate load on the public EESSI CernVM-FS infrastructure. Typically, this means that they should adhere to best practices for CernVM-FS caching in a way that fits the size and nature of their system (e.g. by setting up a private CernVM-FS Stratum 1, a proxy, and/or a properly sized local cache).

System administrators agree to the [Terms of Use](terms_of_use.md) when making EESSI available on the systems they manage.

### 2.9 End-Users

End-Users are individuals that use any of the software provided by the EESSI software stack.

End users agree to the [Terms of Use](terms_of_use.md) when using the software installations provided by EESSI.

> ## 3. Decision-Making

Reviewer 3: Sometimes we use a dash to bind two words sometimes we don't...

> This section applies to the decision making procedure for the Teams discussed in the [Roles and Responsibilities](#roles-and-responsibilities) Section. The Steering Committee has its own Decision-Making procedure (defined in the [Voting by the Steering Communittee](#voting-by-sc) Section).

Reviewer 3: ```suggestion
This section applies to the decision making procedure for the Teams discussed in the [Roles and Responsibilities](#roles-and-responsibilities) Section. The Steering Committee has its own Decision-Making procedure (defined in the [Voting by the Steering Committee](#voting-by-sc) Section).
```

- If a Member is inactive for 6 months and does not reply to communication from the rest of the Steering Committee, the Member may be voted out by the other Steering Committee members. In this case, [regular voting rules](#voting-by-sc) apply.

#### 6.2.2 New members

New Members must be recommended by one or more Steering Committee members.

The Steering Committee will make the recommendation known to the community, and collect feedback on the recommendation.
The Steering Committee then weighs the feedback, and votes on whether to accept the new member.

In the selection process, the Steering Committee will consider the following:

- The Steering Committee ideally consists of an odd number of Members (to reduce chances of a tie when voting);
- No more than 30% of the Members should be employed by commercial entities (to avoid that the project becomes dominated by commercial interests);

> - The number of Members working for the same employer should be limited to 1 (to avoid that the project becomes dominated by a single employer, or a small group of employers);

Reviewer: ```suggestion
- The number of Members working for the same employer should be limited to 1 (to avoid that the project becomes dominated by a single company or institution, or a small group thereof);
```
